lib/instance.js

```diff
@@ -640,16 +641,24 @@ Instance.prototype.save = function(options) {
         self.dataValues[createdAtAttr] = values[self.Model.rawAttributes[createdAtAttr].field || createdAtAttr] = self.Model.$getDefaultTimestamp(createdAtAttr) || now;
       }
 
+      var tableName;
+      if (self.$options.$schema !== self.Model.$schema ||
+        self.$options.$schemaDelimiter !== self.Model.$schemaDelimiter) {
+        tableName = self.Model.schema(self.$options.$schema, self.$options.$schemaDelimiter).getTableName(options);
+      } else {
```

Comment:
But if you clone Model.Instance using Object.create (which is ideal from a resource perspective) it will not matter b/c the Model property of instance is set on the prototype. I guess you could detach the Model from the prototype but this will cause all kinds of memory bloat. I guess you could also create attributes to specifically hold the schema properties on the Instance object and set them but that seems like a worse solution that my current solution that will read the $schema attribute from $options which sets them based on the model the actual instance was built from (https://github.com/sequelize/sequelize/pull/4599/files#diff-a140b60b6a99a8c69f4b5d0d1f1f2bfaL1706) which seems like the expected behavior.